catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceCallable.java

```diff
@@ -243,48 +243,60 @@ public ReliableResourceStatus call() {
 
                 bytesRead.addAndGet(n);
 
-                if (cacheFileOutputStream != null) {
```

Comment:
I don't think we should extract this logic since it is currently protected by the lock and the extracted methods now need to also be protected by the same lock when being executed in this class in the future.  This seems like an unsafe change that should not be part of this PR.